core/src/main/java/org/jsmart/zerocode/core/httpclient/oauth2/OAuth2HttpClient.java

```diff
@@ -0,0 +1,115 @@
+package org.jsmart.zerocode.core.httpclient.oauth2;
+
+import java.util.Map;
+import java.util.Timer;
+
+import org.jsmart.zerocode.core.httpclient.BasicHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.http.client.methods.RequestBuilder;
+
+import java.util.Properties;
+
+/**
+ * @author santhoshTpixler
+ *
+ */
+
+
+/*
+ * Note: This implementation supports the OAuth2.0 with refresh_token and access token
+ * 
+ * Reference: https://tools.ietf.org/html/rfc6749#page-11
+ * 
+ * 	REFRESH TOKEN
+ *  1. The refresh_token, access_token URL, client_id, grant_type and client_secret
+ *     should be generated by the user and stored in the properties file specified by @TargetEnv("host.properties"),
+ *     located at: http-testing/src/test/java/org/jsmart/zerocode/testhelp/tests/OAuth2/OAuth2Test_refreshToken.java
+ *  2. The grant_type should be set to refresh_token in the properties file.
+ *  3. For generating the refresh token REST Client such as Insomnia (https://insomnia.rest/) can
+ *  	be used.
+ *
+ * Note: Postman cannot be used as it does not show the refresh token.
+ *
+ * 	ACCESS TOKEN
+ * 	1. The access_token URL, client_id, grant_type, and client_secret should be generated
+ *     by the user and stored in the properties file specified by @TargetEnv("host.properties"),
+ *     located at: http-testing/src/test/java/org/jsmart/zerocode/testhelp/tests/OAuth2/OAuth2Test_accessToken.java.
+ *  2. The grant_type should be set to client_credentials in the properties file.
+ */
+public class OAuth2HttpClient extends BasicHttpClient {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(OAuth2HttpClient.class);
+
+	/*
+	 * Properties to be fetched from the host.properties
+	 */
+	private static final String CLIENT_ID = "client_id";
+	private static final String CLIENT_SECRET = "client_secret";
+	private static final String REFRESH_TOKEN = "refresh_token";
+	private static final String ACCOUNTS_URL = "accounts_url";
```

Comment:
As per earlier comment:
accounts_url/ACCOUNTS_URL ---> change it to ---> ACCESS_TOKEN_URL/access_token_url

This way it will not create any confusion to the User.